### ISSUE TITLE
DISABLED test_rpc_profiling_remote_record_function (__main__.TensorPipeRpcTestWithSpawn)

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#45067]
https://app.circleci.com/pipelines/github/pytorch/pytorch/216602/workflows/27933844-4828-44d7-bcfa-af3ce27766c0/jobs/7620849/steps

```
Sep 21 04:41:13 ======================================================================
Sep 21 04:41:13 ERROR [1.533s]: test_rpc_profiling_remote_record_function (__main__.TensorPipeRpcTestWithSpawn)
Sep 21 04:41:13 ----------------------------------------------------------------------
Sep 21 04:41:13 Traceback (most recent call last):
Sep 21 04:41:13   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 243, in wrapper
Sep 21 04:41:13     self._join_processes(fn)
Sep 21 04:41:13   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 360, in _join_processes
Sep 21 04:41:13     self._check_return_codes(elapsed_time)
Sep 21 04:41:13   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 396, in _check_return_codes
Sep 21 04:41:13     raise RuntimeError(error)
Sep 21 04:41:13 RuntimeError: Processes 1 exited with error code 10
```

```
Sep 21 04:39:14   test_rpc_profiling_remote_record_function (__main__.TensorPipeRpcTestWithSpawn) ... ERROR:root:Caught exception: 
Sep 21 04:39:14 Traceback (most recent call last):
Sep 21 04:39:14   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_distributed.py", line 246, in wrapper
Sep 21 04:39:14     fn()
Sep 21 04:39:14   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/dist_utils.py", line 70, in new_test_method
Sep 21 04:39:14     return_value = old_test_method(self, *arg, **kwargs)
Sep 21 04:39:14   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/distributed/rpc/rpc_test.py", line 1280, in test_rpc_profiling_remote_record_function
Sep 21 04:39:14     record_function_remote_event.cpu_time_total, remote_ops_time
Sep 21 04:39:14   File "/opt/conda/lib/python3.6/unittest/case.py", line 1227, in assertGreaterEqual
Sep 21 04:39:14     self.fail(self._formatMessage(msg, standardMsg))
Sep 21 04:39:14   File "/opt/conda/lib/python3.6/unittest/case.py", line 670, in fail
Sep 21 04:39:14     raise self.failureException(msg)
Sep 21 04:39:14 AssertionError: 494.69399999999996 not greater than or equal to 573.576
Sep 21 04:39:14 exiting process with exit code: 10
```

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @jjlilley @osalpekar @jiayisuse @mrzzd @agolynski @SciPioneer @H-Huang @cbalioglu @xush6528

**Comment by mrshenli** [comment]
https://app.circleci.com/pipelines/github/pytorch/pytorch/216370/workflows/9a281875-b888-429f-9758-27841741e0de/jobs/7609155/steps

**Comment by rohan-varma** [comment]
It looks like the fundamental issue is not RPC related, but a larger issue w/the profiler: https://github.com/pytorch/pytorch/issues/45160

**Comment by mrshenli** [comment]
closed by #45162

**Comment by mrshenli** [comment]
reopen as the error occurs again. But looks like it only appears after landing this stack (https://github.com/pytorch/pytorch/pull/50029). @rohan-varma does the profiler has any assumption on the `FutureMessage` that `ivalue::Future` violates?

https://app.circleci.com/pipelines/github/pytorch/pytorch/258889/workflows/a6b05e17-ec66-44ab-a133-3376f76a705b/jobs/10077140/steps

https://app.circleci.com/pipelines/github/pytorch/pytorch/258695/workflows/9293a8b7-8a70-42e0-a6e0-b1a62837fb8a/jobs/10067576/steps

https://app.circleci.com/pipelines/github/pytorch/pytorch/258501/workflows/9129a3b3-6751-4e1c-a42d-c17ae657b497/jobs/10056121/steps

**Comment by rohan-varma** [comment]
@mrshenli It should be independent from the future implementation, although the RPC profiling relies on `then()` to ensure the profiling is completed when calling `wait()`, not sure if there are slight differences that can be breaking something. Will investigate

**Comment by rohan-varma** [comment]
Fixed by #57517
